### PR TITLE
fix #32377: `make test-REPL` fails

### LINKED ISSUE #32377: Base.runtests("REPL") fails, but not when run together with some other testset
`Base.runtests("REPL")` (or `Base.runtests(["REPL"])`) fails, but running another testset together with it does not:

```julia
julia> Base.runtests(["REPL"])
[...]
Test Summary: | Pass  Fail  Broken  Total
  Overall     | 1007     1       5   1013
    REPL      | 1007     1       5   1013
    FAILURE

The global RNG seed was 0x975ed040f99437173b72ce7bc2fec875.

Error in testset REPL:
Test Failed at /home/sebastian/Development/julia/usr/share/julia/stdlib/v1.3/REPL/test/replcompletions.jl:107
  Expression: count(isequal("REPL"), c) == 1
   Evaluated: 2 == 1
[...]

julia> Base.runtests(["REPL", "cartesian"])
[...]
Test Summary: | Pass  Broken  Total
  Overall     | 1029       5   1034
    SUCCESS
```

This hapens on Julia 1.0.4, 1.1.1, 1.2.0-rc1, and master.

### PR DESCRIPTION
**Pull request by simeonschaub**
This fixes the issue locally for me.
closes #32377

### CODE REVIEW

File: stdlib/REPL/test/replcompletions.jl [hunk 1]
```diff
@@ -101,6 +101,9 @@ test_scomplete(s) =  map_completion_text(shell_completions(s,lastindex(s)))
 test_bslashcomplete(s) =  map_completion_text(bslash_completions(s,lastindex(s))[2])
 test_complete_context(s) =  map_completion_text(completions(s,lastindex(s),Main.CompletionFoo))
 
+module M32377 end
+test_complete_32377(s) = map_completion_text(completions(s,lastindex(s), M32377))
+
```
Comment (timholy) [inline stdlib/REPL/test/replcompletions.jl:106]: I'm confused by the apparent lack of a module-end statement.

**Comment by simeonschaub** [comment]
Oh, I just saw #32446. It seems like this should be a less controversial change, since I think it still preserves the original intent of the test. Could someone review?

**Comment by simeonschaub** [comment]
@vtjnash You had some thoughts on #32446, do you think this is the right way to go?

File: stdlib/REPL/test/replcompletions.jl [hunk 2]
```diff
@@ -101,6 +101,9 @@ test_scomplete(s) =  map_completion_text(shell_completions(s,lastindex(s)))
 test_bslashcomplete(s) =  map_completion_text(bslash_completions(s,lastindex(s))[2])
 test_complete_context(s) =  map_completion_text(completions(s,lastindex(s),Main.CompletionFoo))
 
+module M32377 end
```
Comment (vtjnash) [inline stdlib/REPL/test/replcompletions.jl:104]: I'd be interested to know if it still works with:
```suggestion
module M32377
    using REPL
end
```
and if not, confirmation that it doesn't represent a real issue.

But yes, this looks like a great solution!

**Review by vtjnash — APPROVED** [review]

**Pull request merged**